whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/client/NrtmRestClient.java

```diff
@@ -82,6 +95,31 @@ public UpdateNotificationFileResponse getNotificationFile(final String source){
                 .get(UpdateNotificationFileResponse.class);
     }
 
+    @Nullable
+    public SnapshotFileResponse getSnapshotFile(final String url){
+        try {
+            final Response response =  client.target(url)
+                    .request(MediaType.APPLICATION_OCTET_STREAM)
+                    .header(HttpHeader.X_FORWARDED_PROTO.asString(), HttpScheme.HTTPS.asString())
+                    .get(Response.class);
+
+            final byte[] payload = response.readEntity(byte[].class);
+            final String[] records = getSnapshotRecords(payload);
```

Comment:
Did you compare the performance? I think loading the whole snapshot file  in one go can increase memory usage quite drastically. We should read it per record, since its a json text sequence read it by a record seprator and pass it to a seperate thread  to process i. You can use the parallelstream to do so

Reply:
the snapshot is compressed in .gz
a json text sequence is inside the .gz

Reply:
[SnapshotImporter] loading snapshot took 3045 ms -> 3 seconds

Reply:
Yes RIPE NONAUTH is very small file so 3sec for it to load is still lot of time. Can you compare for RIPE?  RIPE It doesn't work , means does it throw OOM?

Reply:
yes, even using parallel is throwing OOM
Maybe is because I am testing in a DEV machine which doesn't have too much memory?

Reply:
now I don't have this issue.
I am reading in batches and parallelising those batches. I get the OOM issue if I try to parallelise everything.
I will play with the size of the batches